.github/workflows/ci.yml

```diff
@@ -21,8 +21,46 @@ jobs:
                   python-version: ${{ matrix.python-version }}
             - run: |
                 sudo apt update
-                sudo apt install libdbus-1-dev libgirepository1.0-dev gir1.2-gtk-3.0 gir1.2-notify-0.7 gir1.2-gudev-1.0 graphviz imagemagick x11-apps xserver-xephyr xterm xvfb libpulse-dev lm-sensors
+                sudo apt install libdbus-1-dev libgirepository1.0-dev gir1.2-gtk-3.0 gir1.2-notify-0.7 gir1.2-gudev-1.0 \
+                  graphviz imagemagick x11-apps xserver-xephyr xterm xvfb libpulse-dev lm-sensors libwlroots-dev git ninja-build
                 pip install tox tox-gh-actions
+                # we want wlroots' dependencies only, so remove the packages (these are outdated in ubuntu's repos)
+                sudo dpkg -r --force-depends wlroots libwayland-dev
+                # so we can manually build wayland
+                WAYLAND=1.19.0
+                wget --no-check-certificate https://wayland.freedesktop.org/releases/wayland-$WAYLAND.tar.xz
+                tar -xJf wayland-$WAYLAND.tar.xz
+                cd wayland-$WAYLAND
+                ./configure --disable-documentation
+                make
+                sudo make install
+                cd ..
+                # and wlroots
+                WLROOTS=0.13.0
+                wget --no-check-certificate https://github.com/swaywm/wlroots/archive/$WLROOTS.tar.gz
+                tar -xzf $WLROOTS.tar.gz
+                sudo pip install meson
+                cd wlroots-$WLROOTS
+                meson build
+                ninja -C build
+                sudo ninja -C build install
+                cd ..
+                # and we want git versions of pywayland
+                git clone https://github.com/flacjacket/pywayland
+                cd pywayland
+                pip install -r requirements.txt
+                python pywayland/ffi_build.py
+                python setup.py build
```

Comment:
I _think_ these should be run automatically when `setup.py install` is run, is that not the case?

Reply:
After these are installed, maybe you need to `rm -r` the cloned git directories? I'm not sure if having these packages still available locally is somehow throwing off isort?

Reply:
Ah yeah that would make sense. Hopefully I can avoid this commit altogether though :)